src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -712,6 +712,7 @@ static void destroy_stream_locked(grpc_exec_ctx *exec_ctx, void *sp,
 
   grpc_chttp2_list_remove_stalled_by_transport(t, s);
   grpc_chttp2_list_remove_stalled_by_stream(t, s);
+  grpc_chttp2_list_remove_writing_stream(t, s);
```

Comment:
This is quite dangerous: a different thread could be writing at this point, and so this is likely to race and crash.